File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys
from collections import deque
input = sys.stdin.readline

# n은 정점의 개수, m은 간선의 개수, v는 탐색을 시작할 정점
n, m, v = map(int, input().split())
graph = [[] for i in range(n+1)]
for _ in range(m):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

# 정점 번호가 작은 것을 먼저 방문하기 위해
for line in graph:
    line.sort()

# dfs
visit_dfs = [0 for _ in range(n+1)]
def dfs(v):
    visit_dfs[v] = 1
    print(v, end = ' ')
    for next_v in graph[v]:
        if visit_dfs[next_v] == 0:
            dfs(next_v)
dfs(v)
print()

# bfs
visit_bfs = [0 for _ in range(n+1)]
def bfs(v):
    queue = deque()
    queue.append(v)
    visit_bfs[v] = 1
    while queue:
        v = queue.popleft()
        print(v, end = ' ')
        for next_v in graph[v]:
            if visit_bfs[next_v] == 0:
                queue.append(next_v)
                visit_bfs[next_v] = 1
bfs(v)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.util.*;
import java.io.*;
public class Main {
    static ArrayList<ArrayList<Integer>> list = new ArrayList<>();
    static boolean[] visited_dfs;
    static boolean[] visited_bfs;
    static StringBuilder sb = new StringBuilder();

    static void dfs(int cur) {
        sb.append(cur).append(" ");
        for (int next: list.get(cur)) {
            if (!visited_dfs[next]) {
                visited_dfs[next] = true;
                dfs(next);
            }
        }
    }

    static void bfs(int start) {
        ArrayDeque<Integer> queue = new ArrayDeque<>();
        queue.offer(start);
        while (!queue.isEmpty()) {
            int cur = queue.poll();
            sb.append(cur).append(" ");
            for (int next: list.get(cur)) {
                if (!visited_bfs[next]) {
                    visited_bfs[next] = true;
                    queue.offer(next);
                }
            }
        }
    }

    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        int v = Integer.parseInt(st.nextToken());

        for (int i = 0; i < n + 1; i++) {
            list.add(new ArrayList<>());
        }

        for (int i = 0; i < m; i++) {
            st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            list.get(a).add(b);
            list.get(b).add(a);
        }

        for (int i = 1; i < n + 1; i++) {
            Collections.sort(list.get(i));
        }

        visited_dfs = new boolean[n + 1];
        visited_dfs[v] = true;
        dfs(v);
        sb.append("\n");

        visited_bfs = new boolean[n + 1];
        visited_bfs[v] = true;
        bfs(v);

        System.out.println(sb);
    }
}
